doc/source/guidelines/optional_packages.rst

```diff
@@ -0,0 +1,248 @@
+             "pyside",
+         ]
+         plotting = [
+             "matplotlib",
+             "pyvista",
+         ]
+         qt = [
+             "pyside",
+         ]
+
+         Install ``package-name`` with the optional ``qt`` packages with:
+
+         .. code-block::
+
+            poetry install --extras "plotting qt"
+
+   .. tab:: flit
+
+      .. code-block::
+
+         [project.optional-dependencies]
+         all = [
+             "matplotlib",
+             "pyvista",
+             "pyside",
+         ]
+         plotting = [
+             "matplotlib",
+             "pyvista",
+         ]
+         qt = [
+             "pyside",
+         ]
+
+         Install ``package-name`` with the optional ``qt`` packages with:
+
+         .. code-block::
+
+            pip install package-name --extras=all
+
+   .. tab:: setuptools
+
+      .. code-block:: python
+
+         from setuptools import setup
+
+         setup(
+             ...
+             extras_require={
+                'all': ['matplotlib', 'pyvista', 'pyside'],
+                'plotting': ['matplotlib', 'pyvista'],
+                 'qt': ['pyside'],
+             },
+             ...
+         )
+
+         Install ``package-name`` with the optional ``qt`` packages with:
+
+         .. code-block::
+
+            pip install package-name[qt]
+
+
+Implementing Optional Libraries in Features
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+One of the best ways to implement an optional dependency is to execute a "lazy
+import" at runtime for the feature in question. For example, if your library
+has an optional dependency on ``matplotlib``, you can implement it with:
+
+.. code:: python
+
+   def plot(x, y):
+       """Plot two numpy arrays.
+
+       Parameters
+       ----------
+       x : numpy.ndarray
+           Numpy array sized (n, ).
+       y : numpy.ndarray
+           Numpy array sized (n, ).
+
+       Notes
+       -----
+       This function requires ``matplotlib``.
+
+       """
+       try:
+           import matplotlib.pyplot as plt
+       except ModuleNotFoundError:  # pragma: no cover
+           raise ModuleNotFoundError(
+               "Please install matplotlib to use this feature with:\n\n"
+               "pip install matplotlib"
+           )
+       plt.plot(x, y)
+
+Note that the import statement is within the method and not at the module
+level. Normally this is a bad practice as this can cause runtime errors, but
+for optional features where the user isn't expected to have the library
+installed, this is one of the best ways of handling it. Otherwise, the pyansys
+library might fail to import as the optional package might not be installed.
+
+Also note how this code snippet adds a helpful note `ModuleNotFoundError
+<https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError`_ rather
+than simply letting the error to be raised. This lets the user know that this
+error is expected as the feature relies on an optional dependency.
```

Comment:
```suggestion
Also note how this code snippet adds a helpful `ModuleNotFoundError
<https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError`_ rather
than simply allowing the error to be raised. This lets the user know that this
error is expected because the feature relies on an optional dependency.
```